oraclizeAPI_0.4.25.sol

```diff
@@ -756,164 +2159,2827 @@ contract usingOraclize {
+        dynargs[0] = args[0];
+        dynargs[1] = args[1];
+        dynargs[2] = args[2];
+        dynargs[3] = args[3];
+        return oraclize_token_query(timestamp, datasource, dynargs);
+    }
+
+    function oraclize_token_query(
+        uint timestamp,
+        bytes1 datasource,
+        bytes[4] args,
+        uint gaslimit
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
+        bytes[] memory dynargs = new bytes[](4);
+        dynargs[0] = args[0];
+        dynargs[1] = args[1];
+        dynargs[2] = args[2];
+        dynargs[3] = args[3];
+        return oraclize_token_query(timestamp, datasource, dynargs, gaslimit);
+    }
+
+    function oraclize_token_query(
+        bytes1 datasource,
+        bytes[4] args,
+        uint gaslimit
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
+        bytes[] memory dynargs = new bytes[](4);
+        dynargs[0] = args[0];
+        dynargs[1] = args[1];
+        dynargs[2] = args[2];
+        dynargs[3] = args[3];
+        return oraclize_token_query(datasource, dynargs, gaslimit);
+    }
+
+    function oraclize_token_query(
+        bytes1 datasource,
+        bytes[5] args
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
+        bytes[] memory dynargs = new bytes[](5);
+        dynargs[0] = args[0];
+        dynargs[1] = args[1];
+        dynargs[2] = args[2];
+        dynargs[3] = args[3];
+        dynargs[4] = args[4];
+        return oraclize_token_query(datasource, dynargs);
+    }
+
+    function oraclize_token_query(
+        uint timestamp,
+        bytes1 datasource,
+        bytes[5] args
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
+        bytes[] memory dynargs = new bytes[](5);
+        dynargs[0] = args[0];
+        dynargs[1] = args[1];
+        dynargs[2] = args[2];
+        dynargs[3] = args[3];
+        dynargs[4] = args[4];
+        return oraclize_token_query(timestamp, datasource, dynargs);
+    }
+
+    function oraclize_token_query(
+        uint timestamp,
+        bytes1 datasource,
+        bytes[5] args,
+        uint gaslimit
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
+        bytes[] memory dynargs = new bytes[](5);
+        dynargs[0] = args[0];
+        dynargs[1] = args[1];
+        dynargs[2] = args[2];
+        dynargs[3] = args[3];
+        dynargs[4] = args[4];
+        return oraclize_token_query(timestamp, datasource, dynargs, gaslimit);
+    }
+
+    function oraclize_token_query(
+        bytes1 datasource,
+        bytes[5] args,
+        uint gaslimit
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
+        bytes[] memory dynargs = new bytes[](5);
+        dynargs[0] = args[0];
+        dynargs[1] = args[1];
+        dynargs[2] = args[2];
+        dynargs[3] = args[3];
+        dynargs[4
```

Comment:
Newbie question here... We are comparing `keccak256` because the string compare method would cost more?

Reply:
Yup, as mentioned yesterday.